test_extraction_DK.py

```diff
@@ -4,12 +4,88 @@
 import time
 from extract import extract_to_list
 from torch.utils.data import DataLoader
+from typing import Optional
 
-output_path = "results/epoch1_dev/step8000/carb_dev/"
-dev_gold = "evaluate/OIE2016_dev.txt"
+class RelationExtraction(): 
 
-model_path = "results/model-epoch1-step16000-score1.9768.bin"
+    def __init__(self, model_path: Optional[str] = None):
+        self._bert_config  = "bert-base-multilingual-cased"
+        self._binary = False
+        self._device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self._batch_size = 64
+        self._max_len=64
+        self._num_workers=4
+        self._args = {"bert_config": self._bert_config, "device": self._device, "binary": self._binary}
+        self._bert_model = self.prepare_model(model_path)
+        self._pin_memory = True
+    
 
+# Function for loading a model
+
+    def load_model(self, path):
+        model = utils.get_models(
+            bert_config = self._bert_config,
+            pred_n_labels=3,
+            arg_n_labels=9,
+            n_arg_heads=8,
+            n_arg_layers=4,
+            pos_emb_dim=64,
+            use_lstm=False,
+            device = self._device)
+
+        model.load_state_dict(torch.load(path))
+        model.zero_grad()
+        model.eval()
+
+        return model
+
+# Function for preparing a model. If the path is given, it loads the model. 
+# Otherwise, it creates a folder in the home directory. Then it dowloads the model, saves it to the specified location and loads it.
+
+    def prepare_model(self, model_path): 
+        if model_path is None:
+
+            import urllib.request
+            import os 
+            from pathlib import Path
+            import earthpy as et
+            
+            new_path = os.path.join(et.io.HOME, "Multi2OIE_model")
+            if (not Path(new_path).exists()):
+                os.mkdir(new_path)
+
+            model_path = os.path.join(new_path, "model.bin")
+            urllib.request.urlretrieve(
+                url='https://sciencedata.dk//shared/81ee2688645634814152e3965e74b7f7?download', 
+                filename =  model_path
+                )
+
+        return self.load_model(path = model_path)
+
+    # Function for preparing the dataset
+
+    def prepare_data(self, sents):
+        dataset=EvalDataset(sents, self._max_len, self._bert_config)
+        test_loader = DataLoader( 
+            dataset,
+            self._batch_size,
+            self._num_workers,
+            pin_memory=self._pin_memory
+            )
+        return test_loader
+
+
+    # Function for extracting relations from a given dataset
+ 
+    def extract_relations(self, text):
+        start = time.time()
+        prepared_sent = self.prepare_data(sents = text)
+        extractions = extract_to_list(self._args, self._bert_model, prepared_sent)
+        print("TIME: ", time.time() - start)
+        return extractions
```

Comment:
make the timing optional (can cause a lot of clutter otherwise)

```suggestion
    def extract_relations(self, text: List[str], verbose: bool = False) -> List[Tuple]:
        if verbose:
            start = time.time()
        prepared_sent = self.prepare_data(sents = text)
        extractions = extract_to_list(self._args, self._bert_model, prepared_sent)
        if verbose:
            print("TIME: ", time.time() - start)
        return extractions
```